src/zkp/surjection_proof.rs

```diff
@@ -168,6 +168,77 @@ impl SurjectionProof {
     }
 }
 
+#[cfg(feature = "hashes")]
+impl ::core::fmt::Display for SurjectionProof {
+    fn fmt(&self, f: &mut ::core::fmt::Formatter<'_>) -> ::core::fmt::Result {
+        use bitcoin_hashes::hex::format_hex;
+
+        format_hex(self.serialize().as_slice(), f)
+    }
+}
+
+// TODO: Macrofy (de)serialization
```

Comment:
I think it's fine to leave this all explicit. I'm unconvinced that macro-izing this code improves readability all that much, this is write-once code anyway.

Reply:
Might be worth copying over `serde_util` from here: https://github.com/rust-bitcoin/rust-secp256k1/pull/270